aws/aws-sam-cli-app-templates
# Dotnet31 quickstart s3

*Issue #110 , if available:*

*Description of changes:*
This contribution adds QuickStart S3 project for .NET Core 3.1 since currently there are few QuickStart projects for .NET, I was inspired on the NodeJS QuickStart S3 project and build this one to provide the same boilerplate project for .NET Developers.


Follow these steps to test this PR :

1.	Make sure you’ve .NET Core 3.1.* installed see [Download .NET Core 3.1]( https://dotnet.microsoft.com/download/dotnet/3.1)
2.	Run the following commands

```bash
python3 -m venv
pip install --upgrade pip aws-sam-cli
pip install -r requirements.txt
dotnet --info
sam --version
pytest -vvv tests/integration/build_invoke/test_build_invoke_dotnetcore3_1.py -n 4
pytest -vvv tests/integration/unit_test/test_unit_test_dotnetcore3_1.py -n 4
``` 

And check if the project cookiecutter-aws-sam-quick-start-s3-dotnet was able to run.

**Note:** The unit test of this project seams to works only on Linux or Mac. please do not try to test the commands showed on Windows machine;


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
